app/assets/javascripts/govuk_publishing_components/components/step-by-step-nav.js

```diff
@@ -4,429 +4,483 @@ window.GOVUK = window.GOVUK || {}
 window.GOVUK.Modules = window.GOVUK.Modules || {};
 
 (function (Modules) {
-  'use strict'
-
-  Modules.Gemstepnav = function () {
-    var actions = {} // stores text for JS appended elements 'show' and 'hide' on steps, and 'show/hide all' button
-    var rememberShownStep = false
-    var stepNavSize
-    var sessionStoreLink = 'govuk-step-nav-active-link'
-    var activeLinkClass = 'gem-c-step-nav__list-item--active'
-    var activeStepClass = 'gem-c-step-nav__step--active'
-    var activeLinkHref = '#content'
-    var uniqueId
-
-    this.start = function ($element) {
-      // Indicate that js has worked
-      $element.addClass('gem-c-step-nav--active')
-
-      // Prevent FOUC, remove class hiding content
-      $element.removeClass('js-hidden')
-
-      stepNavSize = $element.hasClass('gem-c-step-nav--large') ? 'Big' : 'Small'
-      rememberShownStep = !!$element.filter('[data-remember]').length && stepNavSize === 'Big'
-      var $steps = $element.find('.js-step')
-      var $stepHeaders = $element.find('.js-toggle-panel')
-      var totalSteps = $element.find('.js-panel').length
-      var totalLinks = $element.find('.gem-c-step-nav__link').length
-      var $showOrHideAllButton
-
-      uniqueId = $element.data('id') || false
-
-      if (uniqueId) {
-        sessionStoreLink = sessionStoreLink + '_' + uniqueId
-      }
+  function Gemstepnav () { }
+
+  Gemstepnav.prototype.start = function ($module) {
+    this.$module = $module[0]
+    this.$module.actions = {} // stores text for JS appended elements 'show' and 'hide' on steps, and 'show/hide all' button
+    this.$module.rememberShownStep = false
+    this.$module.stepNavSize = false
+    this.$module.sessionStoreLink = 'govuk-step-nav-active-link'
+    this.$module.activeLinkClass = 'gem-c-step-nav__list-item--active'
+    this.$module.activeStepClass = 'gem-c-step-nav__step--active'
+    this.$module.activeLinkHref = '#content'
+    this.$module.uniqueId = false
+
+    // Indicate that js has worked
+    this.$module.classList.add('gem-c-step-nav--active')
+
+    // Prevent FOUC, remove class hiding content
+    this.$module.classList.remove('js-hidden')
+
+    this.$module.stepNavSize = this.$module.classList.contains('gem-c-step-nav--large') ? 'Big' : 'Small'
```

Comment:
does bundled gem js have access to govuk-frontent polyfills?

Reply:
That's what I'm currently looking into. The code as it is works fine in most browsers but currently working on some fixes for IE9 and below. IE8 is currently struggling with a lack of support for `indexOf`, I don't know if govuk-frontend supplies a polyfill for that or not. Had some problems with `textContent`, `forEach` and `prepend`, but I've found reasonable workarounds for those.

Reply:
Still not sure whether this has access to govuk-frontend polyfills or not, but since govuk-frontend doesn't support IE8 it doesn't have the polyfills we'd need. I have (perhaps controversially) added some, detailed in the PR description. Happy to discuss.